### PR TITLE
Use new form components on feedback page

### PR DESCRIPTION
**Pull request by tyom**
- Add missing functionality and fix bugs in new form components
- Use new form components on feedback page (Support)

### CODE REVIEW

File: assets/stylesheets/components/form/_error-summary.scss [hunk 1]
```diff
@@ -4,12 +4,20 @@
...
+  &:focus {
+    outline: 3px solid $focus-colour;
+  }
 }
 
 .c-error-summary__heading {
   margin-top: 0;
 }
 
+.c-error-summary__summary:last-child {
```
Comment (teneightfive) [inline assets/stylesheets/components/form/_error-summary.scss]: Great name 👍

File: src/apps/support/controllers.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+const { get, isEmpty } = require('lodash')
+const axios = require('axios')
+
+const logger = require('../../../config/logger')
+const config = require('../../../config')
+
+function postToZenDesk (ticket) {
```
Comment (teneightfive) [inline src/apps/support/controllers.js]: This and `submitToZenDesk ` could be moved to a service.

File: src/apps/support/controllers.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      { id: config.zenBrowser, value: req.body.browser },
+      { id: config.zenService, value: 'datahub_export' },
+    ],
+    tags: [req.body.type],
+  }
+
+  return postToZenDesk(ticket)
+}
+
+async function handleFormSubmission (req, res) {
```
Comment (teneightfive) [inline src/apps/support/controllers.js]: Maybe move this to a middleware that passes through the happy path and add the `renderFeedbackPage ` to the end of the post method.

File: src/apps/support/middleware.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const errors = {
+    messages: {},
+  }
+
+  if (!req.body.title) {
+    errors.messages.title = 'Your feedback needs a title'
+  }
+
+  if (!req.body.feedbackType) {
+    errors.messages.feedbackType = 'You need to choose between raising a bug and leaving feedback'
```
Comment (teneightfive) [inline src/apps/support/middleware.js]: We call _bug_ a _problem_ in the interface

File: src/apps/support/views/feedback.njk [hunk 5]
```diff
@@ -5,8 +5,6 @@
 {% endblock %}
 
 {% block body_main_content %}
-  {{ errorPanel(errors) }}
-
   {% if messages.success.length %}
```
Comment (teneightfive) [inline src/apps/support/views/feedback.njk]: Can this be dropped if being merged after #294

File: src/templates/_macros/form.njk [hunk 6]
```diff
@@ -5,19 +5,25 @@
  # @param {object} errors.messages - A dictionary of errors with field name as key and message as value
  #}
 {% macro ErrorSummary(errors) %}
-  {% if errors.messages | length %}
-    <div class="c-error-summary">
-      <h1 class="c-error-summary__heading heading-medium">This form has errors</h1>
+  {% if errors.messages | length or errors.summary %}
+    <div class="c-error-summary js-ErrorSummary" role="alert">
+      {% if errors.messages | length %}
+        <h1 class="c-error-summary__heading heading-medium">This form has errors</h1>
```
Comment (teneightfive) [inline src/templates/_macros/form.njk]: Do we need to use different language? Surely to a user an error is also a problem submitting.

I'd be tempted to always use _There was a problem submitting this form_.

File: src/templates/_macros/form.njk [hunk 7]
```diff
@@ -29,21 +35,23 @@
  # @param {string} [props.action] - Form action url
  # @param {object} [props.errors] - Object containing form errors
  # @param {boolean} [props.disableFormActions] - Avoid rendering form actions
+ # @param {string} [props.buttonText=Save] - Text for submit button
  #
  # @callback {function} caller - Required inner contents
  #}
 {% macro Form(props) %}
   {% set method = props.method or 'POST' -%}
+  {% set buttonText = props.buttonText or 'Save' %}
```
Comment (teneightfive) [inline src/templates/_macros/form.njk]: Would `Submit` be a better default? You might not always be saving a state.

File: src/templates/_macros/form.njk [hunk 8]
```diff
@@ -29,21 +35,23 @@
...
     {% if props.action %}action="{{ props.action }}"{% endif -%}
   >
     {{ ErrorSummary(props.errors) }}
     {{ CSRF(csrfToken) }}
     {{ caller() }}
-    {% if not form.disableFormActions %}
+    {% if not props.disableFormActions %}
       <div class="c-form-group">
-        <button class="button">Submit</button>
+        <button class="button">{{ buttonText }}</button>
```
Comment (teneightfive) [inline src/templates/_macros/form.njk]: Are we missing the cancel action?

File: src/templates/_macros/form.njk (hunk 7 above)
Comment (tyom) [inline src/templates/_macros/form.njk]: True. Blindly copied.

File: src/templates/_macros/form.njk (hunk 8 above)
Comment (tyom) [inline src/templates/_macros/form.njk]: I was going to address cancel action when I get around building a form that requires it.

File: assets/stylesheets/components/form/_form-errors.scss [hunk 9]
```diff
@@ -1,16 +1,24 @@
 @import "../../settings";
 
-.c-error-summary {
+.c-form-errors {
```
Comment (teneightfive) [inline assets/stylesheets/components/form/_form-errors.scss]: 👍

**Pull request merged**